File: Data Structures & Algorithms/first-missing-positive/submission-6.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        if len(nums) == 1:
            if nums[0] < 0:
                return 1
            else:
                if nums[0] == 1:
                    return nums[0]+1
                else: 
                    return 1

        seen = set()
        n = len(nums)
        for i in range(n):
            seen.add(nums[i])
        
        for i in range(1,n+1):
            if i not in seen:
                return i
        
        return max(nums) + 1

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> seen;
        int n = nums.size();
        if(n==0){
            return 0;
        }
        if(n==1){
            return 1;
        }
        for(int i = 0; i<n; i++){
            seen.insert(nums[i]);
        }

        int max_it = 0;
        for(int x: nums){
            if(seen.find(x-1) != seen.end()){
                continue;
            }
            int cnt = 1;
            int current = x;
            while(seen.find(current + 1) != seen.end()){
                current = current + 1;
                cnt = cnt+1;
            }
            max_it = max(max_it, cnt);
        }
        return max_it;
    }
};
